Automattic/wp-calypso
# Domains: add email verification to domain details

Adds the Icann Verification to Domain Details with a clickable link that leads to editing the email.

To test:
1. Go to Domains / Manage
2. Click on a domain that has unverified email. You should see the Icann Verification card
- Go to Name Servers, try to enable them, and you should see it again.
3. Click on a domain with verified email. It shouldn't be ther.
- Go to Name Servers, try to enable them, the component shouldn't be visible

This is a split from #2916.

Notes:
It was suggested to #2916 that we display the email itself. However, there are several problems with that which would postpone the PR even more, for example the whois endpoint is currently slow and expensive to call and needs to be updated. It also doesn't fail well in case of failure. That's why I decided to go with the earlier suggestion by @ranh and leave the email to be displayed after we resolve the other problems.
